mmdet/models/losses/gfocal_loss.py

```diff
@@ -184,6 +188,24 @@ def forward(self,
                 calculate_loss_func = quality_focal_loss_with_prob
             else:
                 calculate_loss_func = quality_focal_loss
+
+            if isinstance(target, torch.Tensor):
+                # this means that target is already in One-Hot form.
+                assert pred.dim() == target.dim()
+                if target.dim() == 4:
+                    # the target shape with (B,C,W,H), C means classes
+                    pred = pred.permute(0, 2, 3, 1)
+                    target = target.permute(0, 2, 3, 1)
+                # adjust the shape to (N,C)
+                pred = pred.reshape(-1, target.shape[-1])
+                target = target.reshape(-1, target.shape[-1])
+
+                # get category label and quality label info
+                pos_ind, pos_value = torch.max(target, dim=-1)
+                # set bg label
+                pos_ind[~(pos_value > 0)] = target.shape[-1]
+                target = (pos_ind.long(), pos_value.type(pred.dtype))
+
             loss_cls = self.loss_weight * calculate_loss_func(
```

Comment:
Instead of modifying this class, add a new `quality_focal_loss_tensor_target` function to support any dim tensor target